app/models/dataset.rb

```diff
@@ -31,7 +31,12 @@ def initialize(hash)
     @released = hash["released"]
     @licence = hash["licence"]
     @licence_other = hash["licence_other"]
-    @licence_custom = hash["licence_custom"]
+    raw_licence_custom = hash["licence_custom"]
+    @licence_custom ||= if raw_licence_custom && raw_licence_custom.match?(/\[.+\]/)
```

Comment:
Ah, I actually meant for this to be in a method:

```ruby
def licence_custom
  @licence_custom ||= if @raw_licence_custom && @raw_licence_custom.match?(/\[.+\]/)
                          YAML.safe_load(@raw_licence_custom).join("<br />")
                        else
                          @raw_licence_custom
                        end
end
```

But have `@raw_licence_custom = hash["licence_custom"]` in the constructor.

It's a common pattern in Ruby called memoized methods (https://www.justinweiss.com/articles/4-simple-memoization-patterns-in-ruby-and-one-gem/). It means that the work of parsing the YAML can be pushed to the point when it is actually needed and then cached for the lifetime of the instance. If we left it in the constructor, it would be executed every time the class is constructed, even if the value of `licence_custom` is never requested.

Reply:
Tests broke when I tried that, have left a comment to explain why the code is not extracted